### PR TITLE
Update plugin parent POM and dependencies

### PR DESCRIPTION
**Pull request by basil**
**Untested** PR to update the plugin parent POM and dependencies to recent versions. Notably, by depending on Jenkins core 2.321 we can stop bundling Guava and Guice.

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
```
Comment (olamy) [inline .mvn/extensions.xml:1]: no need of incremental here. not sure what is the added value here,

**Review by olamy — CHANGES_REQUESTED** [review]

File: .mvn/extensions.xml (hunk 1 above)
Comment (basil) [reply .mvn/extensions.xml:1]: What is the reason for requesting changes? The presence of incrementals support?

Comment (olamy) [reply .mvn/extensions.xml:1]: what is the added value for this project?

Comment (basil) [reply .mvn/extensions.xml:1]: As stated in the PR description, a slimmer plugin `.hpi` (one that doesn't bundle Guava and Guice).

Comment (olamy) [reply .mvn/extensions.xml:1]: I do not say anything about the rest of the PR but incremental as it's not in the description.

Comment (basil) [reply .mvn/extensions.xml:1]: What is the issue with incrementalizing this plugin?

Comment (olamy) [reply .mvn/extensions.xml:1]: My question is why doing it what is the added value :)

Comment (olamy) [reply .mvn/extensions.xml:1]: it's not a really used plugin, And this change is not part of the PR description and has nothing to do with the rest of the PR

Comment (basil) [reply .mvn/extensions.xml:1]: [JEP-305](https://github.com/jenkinsci/jep/tree/master/jep/305) explains the motivation. Notably, having incrementals automatically published makes it easy to produce builds for users to test, as in [JENKINS-67163](https://issues.jenkins.io/browse/JENKINS-67163).

Comment (olamy) [reply .mvn/extensions.xml:1]: > [JEP-305](https://github.com/jenkinsci/jep/tree/master/jep/305) explains the motivation. Notably, having incrementals automatically published makes it easy to produce builds for users to test, as in [JENKINS-67163](https://issues.jenkins.io/browse/JENKINS-67163).

frankly I don't think we need such complexity here.. not sure we have users asking for a such builds.

Anyway my point is this PR mix up too many unrelated things!

**Pull request closed without merging**